src/components/buttons/default.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+import colors from '../../assets/colors'
+import * as fonts from '../../assets/fonts'
+
+interface ButtonProps extends React.HTMLProps<Button> {
+    state?: ButtonState
+    block?: boolean
+}
+
+export class ButtonState extends String {
+    static Default: ButtonState = "default"
+    static Loading: ButtonState = "loading"
+    static Success: ButtonState = "success"
+    static Failure: ButtonState = "failure"
+}
+
+class Button extends React.Component<ButtonProps, any> {
+    public static defaultProps: ButtonProps
+
+    render(): JSX.Element {
+        const newProps: any = {...this.props}
```

Comment:
Is this possible with TS?

`const { state, ...newProps } = this.props`

I find that to be a little easier to read, not sure if very personal.